models/POP/shell_scripts/cesm2_1/perfect_model.csh

```diff
@@ -0,0 +1,147 @@
+#!/bin/csh
+#
+# Copyright 2020 University Corporation for Atmospheric Research
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License.
+# Please view the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+#=========================================================================
+# This block is an attempt to localize all the machine-specific
+# changes to this script such that the same script can be used
+# on multiple platforms. This will help us maintain the script.
+#=========================================================================
+
+echo "`date` -- BEGIN GENERATE POP TRUE STATE"
+
+# As of CESM2.0, the perfect_model.csh is called by CESM - and has
+# two arguments: the CASEROOT and the DATA_ASSIMILATION_CYCLE
+
+setenv CASEROOT $1
+setenv ASSIMILATION_CYCLE $2
+
+source ${CASEROOT}/DART_params.csh || exit 1
+
+# Python uses C indexing on loops; cycle = [0,....,$DATA_ASSIMILATION_CYCLES - 1]
+# "Fix" that here, so the rest of the script isn't confusing.
+@ cycle = $ASSIMILATION_CYCLE + 1
+
+# xmlquery must be executed in $CASEROOT.
+cd ${CASEROOT}
+setenv CASE                     `./xmlquery CASE                     --value`
+setenv ENSEMBLE_SIZE            `./xmlquery NINST_LND                --value`
+setenv EXEROOT                  `./xmlquery EXEROOT                  --value`
+setenv RUNDIR                   `./xmlquery RUNDIR                   --value`
+setenv ARCHIVE                  `./xmlquery DOUT_S_ROOT              --value`
+setenv TOTALPES                 `./xmlquery TOTALPES                 --value`
+setenv STOP_N                   `./xmlquery STOP_N                   --value`
+setenv DATA_ASSIMILATION_CYCLES `./xmlquery DATA_ASSIMILATION_CYCLES --value`
+setenv TASKS_PER_NODE           `./xmlquery MAX_TASKS_PER_NODE       --value`
+
+#=========================================================================
+# Block 1: Determine time of model state ... from file name
+# of the form "./${CASE}.pop.r.2000-01-06-00000.nc"
+#
+# Piping stuff through 'bc' strips off any preceeding zeros.
+#=========================================================================
+
+cd ${RUNDIR}
+set FILE = `head -n 1 rpointer.ocn.restart`
+# :t is a csh modifier that returns the "tail" of the filename
+# essentially stripping away the filepath that prepends the file
+set FILE = $FILE:t
+# :r is a csh modifier that returns the "root" of the filename 
+# essentially stripping away the file suffix
+set FILE = $FILE:r
+# :e is a csh modifier that returns the "extension" of the filename
+# since we stripped away the .nc in the previous command, the current
+# extension is the datetime in the form YYYY-MM-DD-SSSSS
+set OCN_DATE_EXT = `echo $FILE:e`
```

Comment:
Kevin showed me this ... you can just stack these:
set OCN_DATE_EXT = $FILE:t:r:e
even if you don't want to stack them ... the syntax for OCN_DATE_EXT can be simplified:
set OCN_DATE_EXT = $FILE:e        no need for the extra stuff.